go/vt/vttablet/tabletmanager/init_tablet.go

```diff
@@ -155,7 +155,7 @@ func (agent *ActionAgent) InitTablet(port, gRPCPort int32) error {
 		Hostname:       hostname,
 		PortMap:        make(map[string]int32),
 		Keyspace:       *initKeyspace,
-		Shard:          *initShard,
+		Shard:          shard,
```

Comment:
How is this related?

Reply:
Small topo fix from the last set of commits.

Reply:
I don't understand. You are changing what's stored in the topology, which means there should be a reason for that (which is not explained here) and a test verifying that you store the correct thing, right? I don't understand to which set of commits this is related, and together with the test I think it should be extracted into a separate PR.

Reply:
I pulled this change out into:
https://github.com/youtube/vitess/pull/3417
Will merge and resolve.